packages/Console/src/Application/Executor.php

```diff
@@ -95,18 +99,21 @@ public function run(Section $section, $argv = []): void
         if ($this->matchedSection === null || $this->matchedInput === null) {
             $this->invalidUsage($argv);
 
-            return;
+            return self::EXIT_COMMAND_NOT_FOUND;
         }
 
+        // TODO: what does this condition imply?
         if ($this->beforeAction()->isActionPrevented()) {
-            return;
+            return self::EXIT_OK;
         }
 
+        // TODO: what does this condition imply?
         if (($action = $this->matchedSection->getAction()) === null) {
```

Comment:
@dcole-inviqa not sure what the impications of these conditions are? should they return a non-zero exit status?

Reply:
I thought it was when a sub-command does not match but could be wrong e.g. `ws command sub-command`?

Reply:
No I was wrong its more like if the section is properly configured e.g. see https://github.com/my127/workspace/blob/28edf985214bd03386c9f618dce4fddc7993d413/packages/Console/tests/Test/Application/ApplicationTest.php#L71